src/http/routing.rs

```diff
@@ -67,7 +68,71 @@ macro_rules! routes {
                 #[allow(unused_variables)]
                 let ratelimiting_kind = match *self {
                     $(
-                        Self::$name $({ $($field_name),* })? => $ratelimiting_kind,
+                        Self::$name $({ $($field_name_route),* })? => $ratelimiting_kind,
+                    )+
+                };
+
+                RatelimitingBucket(ratelimiting_kind.map(|r| {
+                    let id = match r {
+                        RatelimitingKind::PathAndId(id) => Some(id),
+                        RatelimitingKind::Path => None,
+                    };
+                    (self.kind(), id)
+                }))
+            }
+
+            #[must_use]
+            pub fn get_owned_route(&self) -> OwnedRoute {
```

Comment:
Rust convention is to name getters without the prefix, but since this is a `&self` -> `Owned` I think the convention would be `to_owned_route`... however that's just ToOwned, so implement `ToOwned` instead.

Reply:
I did not use `toOwned`, because this is not really a clone and should not be treated as one, which is expected by the trait however.

Reply:
With that logic `ToOwned` should not be a thing, but this is exactly what it's for. This is internally just converting `str -> String` which is exactly the point of `ToOwned`.

Reply:
This can not be done using `ToOwned`, because this would require the "copied" object to carry the lifetime with it.

I already tried that sometime, and it makes the whole code a mess.